lib/vsc/install/shared_setup.py

```diff
@@ -156,6 +165,88 @@ def remove_extra_bdist_rpm_files(pkgs=None):
+    """
+    def _mod_setup_py(self, base_dir, external_dir, new_shared_setup):
+        """Modify the setup.py in the distribution directory"""
+
+        # re-copy setup.py, to avoid hardlinks (code from setupstools)
+        dest = os.path.join(base_dir, 'setup.py')
+        log.error('recopying dest %s if hardlinked' % dest)
+        if hasattr(os, 'link') and os.path.exists(dest):
+            # unlink and re-copy, since it might be hard-linked, and
+            # we don't want to change the source version
+            os.unlink(dest)
+            self.copy_file('setup.py', dest)
+
+        fh = open(dest, 'r')
+        code = fh.read()
+        fh.close()
+
+        # look for first line that does someting with vsc.install and shared_setup
+        reg = re.search(r'^.*vsc.install.*shared_setup.*$', code, re.M)
+        if not reg:
+            raise Exception("No vsc.install shared_setup in setup.py?")
+
+        # insert sys.path hack
+        before = reg.start()
+        # no indentation
+        code = code[:before] + NEW_SHARED_SETUP_HEADER_TEMPLATE % (new_shared_setup, VERSION, external_dir) + code[before:]
+
+        # replace 'vsc.install.shared_setup' -> new_shared_setup
+        code = re.sub(r'vsc\.install\.shared_setup', new_shared_setup, code)
+        # replace 'from vsc.install import shared_setup' -> import new_shared_setup as shared_setup
+        code = re.sub(r'from\s+vsc.install\s+import\s+shared_setup', 'import %s as shared_setup', code)
+
+        # write it
+        fh = open(dest, 'w')
+        fh.write(code)
+        fh.close()
+
+    def _add_shared_setup(self, base_dir, external_dir, new_shared_setup):
+        """Create the new shared_setup in distribution directory"""
+
+        ext_dir = os.path.join(base_dir, external_dir)
+        os.mkdir(ext_dir)
+
+        dest = os.path.join(ext_dir, '%s.py' % new_shared_setup)
+        log.error('inserting shared_setup as %s' % dest)
+        try:
+            source_code = inspect.getsource(sys.modules[__name__])
+        except:
+            raise Exception("sdist requires access shared_setup source")
+        try:
+            fh = open(dest, 'w')
+            fh.write(source_code)
+            fh.close()
+        except:
+            raise Exception("Failed to write new_shared_setup source to %s" % dest)
+
+    def make_release_tree(self, base_dir, files):
+        log.error("sdist make_release_tree original base_dir %s files %s" % (base_dir, files))
+        log.error("sdist from shared_setup %s current dir %s" % (__file__, os.getcwd()))
+        if os.path.exists(base_dir):
+            # no autocleanup?
+            # can be a leftover of earlier crash/raised exception
+            raise Exception("base_dir %s present. Please remove it" % base_dir)
+
+        sdist.make_release_tree(self, base_dir, files)
+
+        if __name__ == '__main__':
+            log.error('running shared_setup as main, not adding it to sdist')
+            return
```

Comment:
don't return inline, use `else` instead, and only return below (which is not needed, `return` is there implicitely?